src/main/java/ch/uzh/ifi/hase/soprafs22/service/GameService.java

```diff
@@ -0,0 +1,35 @@
+package ch.uzh.ifi.hase.soprafs22.service;
+
+import ch.uzh.ifi.hase.soprafs22.game.Game;
+import ch.uzh.ifi.hase.soprafs22.game.enums.GameMode;
+import ch.uzh.ifi.hase.soprafs22.game.enums.GameType;
+import ch.uzh.ifi.hase.soprafs22.lobby.LobbyManager;
+import ch.uzh.ifi.hase.soprafs22.repository.UserRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Game Service
+ * This class is the "worker" and responsible for all functionality related to
+ * the game and maps
+ * (e.g., it creates, modifies, deletes, finds). The result will be passed back
+ * to the caller.
+ */
+@Service
+@Transactional
+public class GameService {
+    private final Logger log = LoggerFactory.getLogger(GameService.class);
+    private final UserRepository userRepository;
+
+    @Autowired
+    public GameService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    public Game getInitialGame(GameType gameType, GameMode gameMode){
```

Comment:
Now getting the intial game also creates the game, don't we need to call lobby.getGame() ? Or is that another DT?

Reply:
> Now getting the intial game also creates the game

For my understanding this makes sense doesn't it? Whenever all are ready etc pp, the host of the lobby clicks on start game (or automatically as soon all are ready), and this (click)-event will call the endpoint "/{apiVersion}/game/match/{lobbyId}" which will trigger the creation of a new game with the initial state.

Reply:
But every user has to get the initial game  data => 2/4 games created. also the game does not get stored e.g. it is lost after the call.

If we want to create it with a service we probably should do
- a service to create a game for a lobby Post /{apiVersion}/game/match/{lobbyId}
   - > and after creation lets all other users know
- a service to get game for a lobby Get /{apiVersion}/game/match/{lobbyId}

Reply:
ah ok 👍🏻